umbraco-cloud/upgrades/version-specific-upgrades/migrate-from-umbraco-7-to-8.md

```diff
@@ -0,0 +1,161 @@
+
+See [Step 3](#step-3-setup-custom-code-for-umbraco-8) of this guide, for more detail on this.
+{% endhint %}
+
+## Step 2: Files migration
+
+Before moving on to the next step, make sure that the Umbraco 8 project is no longer running.
+
+The following files/folders need to be copied into the Umbraco 8 project:
+
+* `~/Views` - do **not** overwrite the default Macro and Partial View Macro files, unless changes have been made to these.
+* `~/Media`
+* Any files/folders related to Stylesheets and JavaScripts.
+* Any custom files/folders the Umbraco 7 project uses, that aren't in the `~/Config` or `~/bin`.
+* `~/App_Data/UmbracoForms` - in the case Umbraco Forms was used on the Umbraco 7 site.
+
+Merge the configuration files carefully to ensure any custom settings are migrated while none of the default configurations for Umbraco 8 is overwritten.
+
+### Generating UDA files
+
+1. Run the Umbraco 8 project locally
+    * It **will** give you an error on the frontend as none of the Template files have been updated yet.
+2. Open the command line tool in the `~/data` folder on the Umbraco 8 project.
+3. Generate UDA files by running the following command: `echo > deploy-export`.
+    * Once a `deploy-complete` marker is added to the `~/data` folder, it is done.
+4. Check `~/data/revision` to ensure all the UDA files have been generated.
+5. Run `echo > deploy` in the `~/data` folder to make sure everything checks out with the UDA files that were generated.
+
+Running the `echo > deploy` command will generate a new marker. Move forward with the migration based on the marker:
+
+* `deploy-failed`
+  * Something failed during the check
+  * Run `echo > deploy-clearsignatures` followed by `echo > deploy` to clear up the error
+* `deploy-complete`
+  * Everything checks out: Move on to the next step
+
+## Step 3: Setup custom code for Umbraco 8
+
+Umbraco 8 is different from Umbraco 7 in many ways. This means that in this step, all custom code, controllers, and models need to be reviewed and rewritten for Umbraco 8.
+
+### Example of changes that need to be made
+
+One of the changes made, is how published content is rendered through Template files. Due to this, it will be necessary to update **all** Template files (`.cshtml`) to reflect these changes.
+
+Read more about these changes in the [IPublishedContent section of the Documentation](https://docs.umbraco.com/umbraco-cms/reference/querying/ipublishedcontent).
+
+* Template files need to inherit from `Umbraco.Web.Mvc.UmbracoViewPage<ContentModels.HomePage>` instead of `Umbraco.Web.Mvc.UmbracoTemplatePage<ContentModels.HomePage>`.
+* Template files need to use `ContentModels = Umbraco.Web.PublishedModels` instead of `ContentModels = Umbraco.Web.PublishedContentModels`.
+* `@Model.Value("propertyAlias")` replaces `@Umbraco.Field("propertyAlias")`.
+* `@Model.PropertyAlias` replaces `@Model.Content.PropertyAlias`.
+
```

Comment:
I also had to make this change in the template to view the frontend:
`Model.Value<IPublishedContent>("propertyAlias");` replaces `Model.Content.GetPropertyValue<IPublishedContent>("propertyAlias");`

Reply:
Ah, sure! I didn't test update the rendering, as that's not really something we document anyways, so must have missed this one.

Does it replace the third bullet, or is just a new bullet entirely? 🤔